Library/Homebrew/cask/lib/hbc/artifact/moved.rb

```diff
@@ -65,10 +65,11 @@ def move_back(skip: false, force: false, command: nil, **options)
         ohai "Backing #{self.class.english_name} '#{target.basename}' up to '#{source}'."
         source.dirname.mkpath
 
+        # We need to preserve extended attributes between copies.
         if target.parent.writable?
-          FileUtils.cp_r(target, source)
+          command.run!("/bin/cp", args: ["-pR", target, source])
```

Comment:
The only difference here is `sudo`, so maybe make this a variable which is set by `target.parent.writable?`.